### DISCUSSION TITLE
Homebrew tap not able to release to a Github repo in a Github Organisation

### DISCUSSION THREAD

**Discussion by Pradumn99** [goreleaser/goreleaser#3000] [Q&A] [answered]
Hi Team ,

Can anyone help me this issue. I want to publish homebrew tap to a repo which is in an organisation. But the generated URL is " https://api.github.com/repos/ABC-organisation/XYZ-Repo/contents/Formula/pqr.rb: 404 Not Found []", where ABC, XYZ and pqr are arbitrary name. 

Is there a way we can manipulate this URL created ?
Or use goreleaser's homebrew tap to publish to repo inside an organisation.

Thanks in advance

**Comment by Pradumn99** [comment]
@caarlos0 can you guide me on this.

**Comment by caarlos0** [comment, marked as answer]
you need to have the repo already created, and the token must have access to it.

goreleaser itself publishes its own formula to an organization.

more info: https://goreleaser.com/customization/homebrew/?h=brew

---

###### please make sure to reply in this thread instead of a new one, thanks!

**Reply by Pradumn99** [reply]
The repo already exists and the token has access to the repo as well. The issue is the URL that is generated as a part Homebrew. tap, which gives a 404, repo not found error.

**Reply by caarlos0** [reply]
isn't it a typo somewhere?

if not, I'm not sure, but maybe the `Formula` folder needs to be created beforehand too

**Reply by Pradumn99** [reply]
Thanks for your reply, Just wanted to tell you that we have an organisation in github and inside  that we have a repo which has the Formula Folder. 

The issue that I face is, the URL generated is not right. any way to manipulate the URL in this homebrew section of goreleaser.

**Reply by Pradumn99** [reply]
I am not sure but can we access the repository inside a github org with "https://api.github.com/repo/OrgName/RepoName*" url.

In release section of goreleaser, the url is created as "https://github.com/OrgName/RepoName/*" which works all fine.

**Reply by caarlos0** [reply]
what do you mean by "not right"? the org or repo name are incorrect? if so, are you setting `brews.tap` properly? 

https://goreleaser.com/customization/homebrew/?h=brew

if thats not it, please provide a reproducible. Its hard to guess whats wrong without seeing the config/etc.

**Reply by Pradumn99** [reply]
Hi @caarlos0, actually by "not right", I meant that the Github URL to which we want to publish the Tap is not right since it is not able to locate the repo. It displays that there is no Repo of that name. But in actual the repo exist within a github organisation.

**Reply by Pradumn99** [reply]
It is more like I am trying to publish the homebrew recipe Tap to a private github organisation and a private repo in that organisation.

**Reply by caarlos0** [reply]
probably your github token does not have enough permissions.

**Reply by Pradumn99** [reply]
@caarlos0 Thanks a lot for your help. Figured out and issue is resolved. It is working all fine.

One more help, that if you could right now is that when the homebrew tap recipe is created and published on the repo, I am getting "**bottle: uneeded**" in it. I do not want that in the recipe file of the homebrew. is there a way I can suppress this.

**Reply by caarlos0** [reply]
not possible, no
